src/components/BarChart/BarChart.jsx

```diff
@@ -335,6 +358,46 @@ const BarChart = createClass({
 			? yAxisTooltipDataFormatter
 			: yAxisFinalFormatter;
 
+		if (_.isEmpty(data) || width < 1 || height < 1 || isLoading) {
+			const emptyMessageBodyProp = _.get(getFirst(this.props, BarChart.EmptyMessageBody), 'props');
+			const emptyMessageTitleProp = _.get(getFirst(this.props, BarChart.EmptyMessageTitle), 'props', {children: 'You have no data.'});
+
+			return (
+				<EmptyStateWrapper
+					isEmpty={_.isEmpty(data)}
+					isLoading={isLoading}
+				>
+					<EmptyStateWrapper.EmptyMessageBody {...emptyMessageBodyProp}/>
+					<EmptyStateWrapper.EmptyMessageTitle {...emptyMessageTitleProp}/>
+					<svg
+						{...omitProps(passThroughs, BarChart)}
```

Comment:
I feel like in this case and cases like this, even though the `Wrapper` is hierarchically the root element, my expectation as a user of the component would be that the root element is the `<svg>`. It feels like kind of an exception to the pattern.